libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -262,88 +269,100 @@ private void syncUp(SyncState sync, SyncUpdateCallback callback) throws Exceptio
 		String soupName = sync.getSoupName();
 		SyncOptions options = sync.getOptions();
 		List<String> fieldlist = options.getFieldlist();
-		QuerySpec querySpec = QuerySpec.buildExactQuerySpec(soupName, LOCAL, "true", 2000); // XXX that could use a lot of memory
```

Comment:
New approach on sync up. Getting all the ids (looping through pages if needed) then doing retrieve and save to server in a loop one record at a time. (unrelated to the story)